galaxy_ng/app/access_control/access_policy.py

```diff
@@ -213,6 +243,56 @@ def has_distribution_repo_perms(self, request, view, action, permission):
 
         return False
 
+    def signatures_not_required_for_repo(self, request, view, action):
+        """
+        Validate that collections are being added with signatures to approved repos
+        when signatures are required.
+        """
+        print("______--------____------__------__-----__--------")
+        repo = view.get_object()
+        repo_version = repo.latest_version()
+
+        src_pipeline = repo.pulp_labels.get("pipeline", None)
+
+        if settings.GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL:
+            if src_pipeline == "approved":
+                return True
+        else:
+            return True
```

Comment:
I don't understand this block, both branches returns True, are those redundant?
logically the same

```suggestion
        if settings.GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL:
            return True
```

also, why is it returning True when that setting is set? no need for the extra checks below?

Reply:
I was thinking to skip the check if you're copying from an approved repo, because those should theoretically have signatures already, but I guess it makes more sense to just check the destinations.